### ISSUE TITLE
Agent bytecode instrumentation wrong for Apache HttpClient

### ISSUE THREAD

**Issue by jhsheets** [Microsoft/ApplicationInsights-Java#276]
The HttpClassDataProvider.java file has the following constant defined:

`private final static String HTTP_CLIENT_METHOD_42_SIGNATURE = "(Lorg/apache/http/HttpHost;Lorg/apache/http/HttpRequest;Lorg/apache/http/protocol/HttpContext;)Lorg/apache/http/HttpResponse;";`

This isn't the actual method signature, and it's causing errors. The actual signature should be:
`private final static String HTTP_CLIENT_METHOD_42_SIGNATURE = "(Lorg/apache/http/HttpHost;Lorg.apache.http.client.methods.HttpUriRequest;Lorg/apache/http/protocol/HttpContext;)Lorg/apache/http/HttpResponse;";
`

Without this fix, libraries that use HttpClient are blowing up. Here's an example using agent on a webapp using Amazon's S3 java library (which depends on HttpClient):

```
Caused by: java.lang.VerifyError: (class: org/apache/http/impl/client/AbstractHttpClient, method: execute signature: (Lorg/apache/http/HttpHost;Lorg/apache/http/HttpRequest;Lorg/apache/http/protocol/HttpContext;)Lorg/apache/http/HttpResponse;) Incompatible object argument for function call
at com.amazonaws.http.HttpClientFactory.createHttpClient(HttpClientFactory.java:106)
at com.amazonaws.http.AmazonHttpClient.<init>(AmazonHttpClient.java:193)
at com.amazonaws.AmazonWebServiceClient.<init>(AmazonWebServiceClient.java:119)
at com.amazonaws.AmazonWebServiceClient.<init>(AmazonWebServiceClient.java:103)
at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:365)
at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:347)
```

**Comment by gupele** [comment]
Hi,

Thank you for your input!

We will need to look into this, meanwhile you can disable the built-in 'HTTP' instrumentation using the AI-Agent.xml configuration file.
If that is still something you need to monitor, you can add it 'manually' in that file with the correct signature

**Comment by jhsheets** [comment]
I tried disabling the instrumentation once, but it continued to throw the
exception. It's possible I didn't disable it properly, but I thought I did
it right.

**Comment by dhaval24** [comment]
Fixed in #449
